src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/CalculationTransactionalService.kt

```diff
@@ -52,25 +51,22 @@ class CalculationTransactionalService(
       ?: throw IllegalStateException("User is not authenticated")
 
   /*
-   * There are 4 stages of full validation:
-   * 1. validate user input data and the raw sentence and offence data from NOMIS
-   * 2. Validate the transformation of the raw Booking (NOMIS offence data transformed into a Booking object)
-   * 3. Run the calculation and catch any errors thrown by the calculation algorithm
-   * 4. Validate the post calculation Booking (The Booking is transformed during the calculation). e.g. Consecutive sentences (aggregates)
+   * There are 3 stages of full validation:
+   * 1. validate user input data, the raw sentence and offence data from NOMIS and the raw Booking (NOMIS offence data transformed into a Booking object)
+   * 2. Run the calculation and catch any errors thrown by the calculation algorithm
+   * 3. Validate the post calculation Booking (The Booking is transformed during the calculation). e.g. Consecutive sentences (aggregates)
    *
    * activeDataOnly is only used by the test 1000 calcs functionality
    */
   fun fullValidation(prisonerId: String, calculationUserInputs: CalculationUserInputs, activeDataOnly: Boolean = true): List<ValidationMessage> {
     val sourceData = prisonService.getPrisonApiSourceData(prisonerId, activeDataOnly)
-    var messages = validationService.validateSourceData(sourceData) // Validation stage 1 of 4
+    var messages = validationService.validateBeforeCalculation(sourceData, calculationUserInputs) // Validation stage 1 of 3
     if (messages.isNotEmpty()) return messages
-
     val booking = bookingService.getBooking(sourceData, calculationUserInputs)
-    messages = validationService.validateBookingBeforeCalculation(booking) // Validation stage 2 of 4
-    if (messages.isNotEmpty()) return messages
     try {
-      val bookingAfterCalculation = calculationService.calculate(booking) // Validation stage 3 of 4
-      validationService.validateBookingAfterCalculation(bookingAfterCalculation) // Validation stage 4 of 4
+      // TODO Doesn't look like this calculate call actually throws any CrdCalculationValidationException, so this whole try catch block can be removed
```

Comment:
One for @ldlharper to advise on, I think we can remove this try catch block.. (separate ticket if we do it)

Reply:
Do we no longer throw these exceptions for validation? There were a few, remand overlapping with sentence and extinguished sentences.

Reply:
These rules are fired by the post calc validation, which used to throw exceptions.. but now just returns messages. Will talk about tjis separately though